extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/handler/GenericThingHandler.java

```diff
@@ -109,29 +127,44 @@ protected ChannelState createChannelState(ChannelConfig channelConfig, ChannelUI
         ChannelState state = new ChannelState(channelConfig, channelUID, valueState, this);
 
         // Incoming value transformations
-        TransformationServiceProvider transformationServiceProvider = this.transformationServiceProvider;
-        if (transformationServiceProvider != null && StringUtils.isNotBlank(channelConfig.transformationPattern)) {
-            state.addTransformation(
-                    new ChannelStateTransformation(channelConfig.transformationPattern, transformationServiceProvider));
-        }
+        String[] transformations = channelConfig.transformationPattern.split("\n");
+        Stream.of(transformations).filter(t -> StringUtils.isNotBlank(t))
+                .map(t -> new ChannelStateTransformation(t, transformationServiceProvider))
+                .forEach(t -> state.addTransformation(t));
 
         return state;
     }
 
     @Override
     public void initialize() {
+        List<ChannelUID> configErrors = new ArrayList<>();
         for (Channel channel : thing.getChannels()) {
             final ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
             if (channelTypeUID == null) {
                 logger.warn("Channel {} has no type", channel.getLabel());
                 continue;
             }
             final ChannelConfig channelConfig = channel.getConfiguration().as(ChannelConfig.class);
-            ChannelState channelState = createChannelState(channelConfig, channel.getUID(),
-                    ValueFactory.createValueState(channelConfig, channelTypeUID.getId()));
-            channelStateByChannelUID.put(channel.getUID(), channelState);
+            try {
+                Value value = ValueFactory.createValueState(channelConfig, channelTypeUID.getId());
+                ChannelState channelState = createChannelState(channelConfig, channel.getUID(), value);
+                channelStateByChannelUID.put(channel.getUID(), channelState);
+                StateDescription description = value.createStateDescription(channelConfig.unit,
+                        StringUtils.isBlank(channelConfig.commandTopic));
+                stateDescProvider.setDescription(channel.getUID(), description);
+            } catch (IllegalArgumentException e) {
+                logger.warn("Channel configuration error", e);
+                configErrors.add(channel.getUID());
+            }
         }
 
-        super.initialize();
+        // If some channels could not start up, put the entire thing offline and display the channels
```

Comment:
is this the way you deal with the new channel config description?

Reply:
I have added this, because of the channel type id renaming (Switch -> switch etc). People on the forum report it as a bug that the Thing goes offline, and I'm now showing why (the channel has to be recreated).